src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -52,6 +53,15 @@ util::geo::I32Box SpatialJoinAlgorithms::libspatialjoinParse(
       const auto& p = id.getGeoPoint();
       parser.parsePoint(util::geo::DPoint(p.getLng(), p.getLat()), row,
                         leftOrRightSide);
+    } else if (id.getDatatype() == Datatype::LocalVocabIndex) {
+      const auto& literalOrIri = result->localVocab()
+                                     .getWord(id.getLocalVocabIndex())
+                                     .asLiteralOrIri();
+      if (literalOrIri.isLiteral()) {
+        const auto& wkt = std::string(
+            asStringViewUnsafe(literalOrIri.getLiteral().getContent()));
+        parser.parseWKT(wkt.c_str(), row, leftOrRightSide);
+      }
```

Comment:
This code should be testable, 
you can just for example create a `VALUES`clause as one of the inputs that hold geoliterals that are not part of the original index.